### PR TITLE
eslint [nfc]: Give comments more room to breathe; regularize indentation.

### PR DESCRIPTION
**Pull request by gnprice**
Things I noticed in the course of making the edits in 88cafe4df, that can perhaps make future edits to this config a little easier.

This is `eslint [nfc]` because it makes no functional changes even *to* the ESLint config; it just touches comments and whitespace.

### CODE REVIEW

**Comment by gnprice** [comment]
@ray-kraesig , one other thought that came to mind looking at this: what's the reason why the overrides on `files: ["*.js"]` are there rather than in the main `rules` block? A comment says they need to be, but I don't recall if you mentioned at the time what went wrong if they're normal rules, and the commit 245ea9c4cdf8fb9459ba64dff3e71dab89b36c75 and PR #3725 don't say.

File: .eslintrc.yaml [hunk 1]
```diff
@@ -76,47 +91,74 @@ rules:
...
   - patterns: ['**/__tests__/**']
+
   import/no-cycle: off  # This would be nice to fix; but isn't easy.
 
-  # Jest
+
+  #
+  # Jest; plugin `jest`.
+  #
+
```
Comment (ray-kraesig) [inline .eslintrc.yaml]: This doesn't really look like it's supposed to be a sort of section header.

I'd suggest something like

```
# ===================================================================
# Jest

# This rule could be useful if...
```

or, with somewhat more brevity,

```
####
# Jest

# This rule could be useful if...
```

or even the Markdown-style

```
# ## Jest
#
# This rule could be useful if...
```

... although personally I'd include the `#` in all cases, not just the final one.

File: .eslintrc.yaml [hunk 2]
```diff
@@ -76,47 +91,74 @@ rules:
...
-  # React Native.
+
+  #
+  # React Native; plugin `react-native`.
   #
+
   # This plugin isn't included in the airbnb config, and it doesn't
   # itself enable any rules by default, so this config is it.
-  #
+
```
Comment (ray-kraesig) [inline .eslintrc.yaml]: ?

**Comment by gnprice** [comment]
I've just pushed a refreshed version of this: rebased across the changes since it was originally written, addressing Ray's comment above at https://github.com/zulip/zulip-mobile/pull/3971#discussion_r393253375 , and adding a few new commits.

**Comment by chrisbobbe** [comment]
Looks great, thanks! Merged.

**Pull request merged**